### PR TITLE
Put some order on the credential service

### LINKED ISSUE #666: Inserting existing scopes throws "one or more scopes don't exists"
When trying to insert existing scopes, you get an error "one or more scopes don't exists".

flow - 
Add some scopes > try to add the same scopes again > Error.

problem could be the syntax / the error throwing itself (based on wanted behavior)

### PR DESCRIPTION
**Pull request by XVincentX**
The following pull request will put some orders in our credentials->scopes parts.

In particular:

* There was some confusion on `scope` and `scopes` parameters. The first one was misused and, moreover, undocumented. I've removed it and made sure all the occurrences are using always `scopes` as an array.
* The message when creating the same scope multiple times was broken. Fixed with the correct one.

Connect #666 
Closes #666

### CODE REVIEW

**Comment by kevinswiber** [comment]
Looks okay to me.  I would like @DrMegavolt to take a look, as well.  In general, I have a lot of concerns about our modifying our Identity Server functionality.  Changes always seem to have big cascading effects.

**Comment by XVincentX** [comment]
@kevinswiber 

I know these are changes hard to review and maybe a bit harmful and I apologise for that, but I really want to fix these things when I end up spending half of a day to solve a simple bug because the code is mentally hard to follow and goes in unpredictable paths. 

Spend again half a day to rebuild the "flow-map" of the code on the scopes next time I'll have to work is something I definitely would like to avoid.

File: test/services/credential-service/credentials.test.js [hunk 1]
```diff
@@ -158,13 +158,13 @@ describe('Credential service tests', () => {
...
       secret: 'password',
-      scopes: 'someScope',
+      scopes: ['someScope'],
       someProperty: 'propVal'
     };
 
     before(() => {
       Object.assign(config.models.credentials.properties.oauth2.properties, {
-        scopes: { type: 'string' },
+        scopes: { type: 'array', items: { tyep: 'string' } },
```
Comment (DrMegavolt) [inline test/services/credential-service/credentials.test.js]: typo `tyep` , and if it works with this error I would just remove the line, since it is not needed

**Comment by DrMegavolt** [comment]
Except for the typo, I also see `grabScopesAndExecute` method is now not covered by tests. not sure what it does but it would be good to restore coverage 

other than that looks ok

File: test/services/credential-service/credentials.test.js (hunk 1 above)
Comment (XVincentX) [reply test/services/credential-service/credentials.test.js]: Correct, this line is useless and can be removed. Thanks for spotting this.
